### PR TITLE
Adds create-release workflow and supporting scripts

### PR DESCRIPTION
**Pull request by carlo-808**
## Proposed Release Notes
* Adds `Create Release` github actions workflow to create a release for `patch` or `minor` releases
* Updates agent version and adds release tags
## Links
Closes: #585

### CODE REVIEW

File: bin/create-release.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    if (!passesWorkflowChecks) {
+      process.exit(1)
+    }
+  } catch (err) {
+    console.log(err)
+
+    process.exit(1)
+  }
+
+  exec(`npm version ${releaseType} && git push origin ${branch} && git push --tags`, (err, stdout) => {
```
Comment (michaelgoin) [inline bin/create-release.js:54]: Not sure what the output looks like but if nothing obvious, we may want to write a message before executing this step.

File: bin/github.js [hunk 2]
```diff
@@ -13,94 +10,110 @@ const octokit = new Octokit({
...
-
-  return result.data
-}
-
-async function getTagByName(name) {
-  const perPage = 100
+class Github {
+  constructor(repoOwner, repository) {
+    // Default to node agent repo for now
+    this.repoOwner = repoOwner ? repoOwner : 'newrelic'
```
Comment (michaelgoin) [inline bin/github.js:16]: could be simplified to: `this.repoOwner = repoOwner || 'newrelic'`

File: bin/github.js [hunk 3]
```diff
@@ -13,94 +10,110 @@ const octokit = new Octokit({
...
-    owner: repoOwner,
-    repo: repository
-  })
-
-  return result.data
-}
-
-async function getTagByName(name) {
-  const perPage = 100
+class Github {
```
Comment (michaelgoin) [inline bin/github.js:13]: Given the change to be a class now... I believe generate-release-notes.js will be broken as it has not been updated to use the class.

**Review by michaelgoin — CHANGES_REQUESTED** [review]
Since generate-release-notes will be broken, requesting changes on this PR.

Minor comments don't need to be addressed unless desired.

An additional note (can be handled separately): we are getting to enough related scripts that it likely is starting to make sense to organize these into an additional folder at some point.

File: bin/check-workflow-run.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    const latestRun = await github.getLatestWorkflowRun('ci-workflow.yml', branch)
+
+    if (latestRun === undefined) {
+      console.log('No ci workflow run found.')
+    } else {
+      console.log('CI workflow run details: ', JSON.stringify(formatRun(latestRun)))
+      if (latestRun.status === 'completed' && latestRun.conclusion === 'success') {
+        results.ci = true
```
[the remark this answers is not in the document]
Comment (carlo-808) [reply bin/check-workflow-run.js:37]: Sounds good, not changing this at this time.

**Review by michaelgoin — APPROVED** [review]

**Pull request merged**